src/client/eppo-client.ts

```diff
@@ -152,6 +144,12 @@ export default class EppoClient {
     this.banditVariationConfigurationStore = banditVariationConfigurationStore;
   }
 
+  /** Sets the EventDispatcher instance to use when tracking events with {@link track}. */
+  // noinspection JSUnusedGlobalSymbols
+  setEventDispatcher(eventDispatcher: EventDispatcher) {
+    this.eventDispatcher = eventDispatcher;
+  }
```

Comment:
i really don't love these mutable setter APIs we have in the `EppoClient`, as a fan of immutable data types, i'd much rather do it like that, but I'm following the existing pattern here

Reply:
Agreed, anything that won't toggle after instantion should be constructor options and immutable

Reply:
that would be a bigger refactoring, so maybe we can consider for a future major release